recipes/xerces-c/all/conanfile.py

```diff
@@ -32,16 +37,70 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
-    def validate(self):
-        if self.options.char_type == "wchar_t" and self.settings.os != "Windows":
+    def _validate(self, option: str, value: str, os: Union[str, Tuple[str]]):
+        """
+        Validate that the given `option` has the required `value` for the given `os`
+        If not raises a ConanInvalidConfiguration error
+
+        :param option: the name of the option to validate
+        :param value: the value that the `option` should have
+        :param os: either a single string or a tuple of strings containing the
+                   OS(es) that `value` is valid on
+        """
+        if ((isinstance(os, str) and self.settings.os != os) or self.settings.os not in os) \
+             and getattr(self.options, option) == value:
             raise ConanInvalidConfiguration(
-                "Option 'char_type=wchar_t' is only supported in Windows"
+                f"Option '{option}={value}' is only supported on {os}"
             )
 
+    def validate(self):
+        self._validate("char_type", "wchar_t", "Windows")
+        self._validate("network_accessor", "winsock", "Windows")
+        self._validate("network_accessor", "cfurl", "Macos")
+        self._validate("network_accessor", "socket", ("Linux", "Macos"))
+        self._validate("network_accessor", "curl", ("Linux", "Macos"))
+        self._validate("transcoder", "macosunicodeconverter", "Macos")
+        self._validate("transcoder", "windows", "Windows")
+        self._validate("mutex_manager", "posix", ("Linux", "Macos"))
+        self._validate("mutex_manager", "windows", "Windows")
+
+    def requirements(self):
+        if "icu" in (self.options.transcoder, self.options.message_loader):
+            self.requires("icu/69.1")
+
+    def _set_default_option(self, option: str, default_value, os: str = ""):
+        """
+        Set the given `option` to the `default_value` if it is empty (i.e. it has not been set by the user)
```

Comment:
I didn't think the precaution of looking for options already set by the user is necessary because `config_options` is executed before the actual assignment of options. https://docs.conan.io/en/latest/reference/conanfile/methods.html#configure-config-options

Reply:
Oh, you're totally right. I must have overlooked that. Thanks for catching this!